### ISSUE TITLE
Since 0.8.3, aws-crt-{cpp,python} tests are broken

### ISSUE THREAD

**Issue by fossdd** [awslabs/aws-c-http#480]
### Describe the bug

By building and testing aws-crt-cpp or aws-crt-python with awc-c-http 0.8.3, 3 tests fail:

Seems to affect 0.8.3 and higher, 0.8.2 is unaffected, so likely introduced by #471 or #469

I believe more dependents of aws-c-http are affected, but on Alpine Linux, we're only using them.

### Expected Behavior

That the tests of aws-crt-cpp and aws-crt-python succeed.

### Current Behavior

The logs of the failing aws-crt-cpp are below:

<details>

<summary>155/175: TestCreatingImdsClient</summary>

```
[...]
155/175 Test #112: TestCreatingImdsClient ............................................Subprocess aborted***Exception:   0.23 sec
[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [aws-c-common] - static: libnuma.so failed to load
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - searching process and loaded modules
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - did not find aws-lc symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - did not find boringssl symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - did not find libcrypto 1.0.2 symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - found static libcrypto 1.1.1 HMAC symbols
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - found static libcrypto 1.1.1 EVP_MD symbols
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [libcrypto_resolve] - Compiled with libcrypto OpenSSL 3.3.1 4 Jun 2024, linked to libcrypto OpenSSL 3.3.1 4 Jun 2024
[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [tls-handler] - static: Initializing TLS using s2n.
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [tls-handler] - ctx: Based on OS, we detected the default PKI path as /etc/ssl/certs, and ca file as /etc/ssl/certs/ca-certificates.crt
[ERROR] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/sys_vendor' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[ERROR] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/product_name' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [aws-c-common] - id=0x74a9c1d91f40: virtualization vendor detected as ""
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [aws-c-common] - id=0x74a9c1d91f40: virtualization product name detected as " "
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "c5n.18xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "c5n.9xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "c5n.metal".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "p4d.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "p4de.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "p5.48xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "trn1n.32xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [S3General] - id=0x74a9c1d7c0b0: adding platform entry for "trn1.32xlarge".
[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [event-loop] - id=0x74a9c1d7dc70: Initializing edge-triggered epoll
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [event-loop] - id=0x74a9c1d7dc70: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [event-loop] - id=0x74a9c1d7dc70: pipe descriptors read 6, write 7.
[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [event-loop] - id=0x74a9c1d7dc70: Starting event-loop thread.
malloc failed to allocate memory[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [dns] - id=0x74a9c1ddcca0: Initializing default host resolver with 8 max host entries.
[INFO] [2024-07-31T10:09:00Z] [000074a9c1d73b38] [event-loop] - id=0x74a9c1d7dc70: main loop started
[INFO] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [channel-bootstrap] - id=0x74a9c1d7e540: Initializing client bootstrap with event-loop group 0x74a9c1db1620
[TRACE] [2024-07-31T10:09:00Z] [000074a9c1d73b38] [event-loop] - id=0x74a9c1d7dc70: subscribing to events on fd 6
[DEBUG] [2024-07-31T10:09:00Z] [000074a9c2913cb0] [channel-bootstrap] - id=0x74a9c1d7e540: acquiring bootstrap reference
[INFO] [2024-07-31T10:09:00Z] [000074a9c1d73b38] [event-loop] - id=0x74a9c1d7dc70: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000074a9c1d73b38] [event-loop] - id=0x74a9c1d7dc70: waiting for a maximum of 100000 ms

        Start 163: Mqtt5to3AdapterWSConnectionMinimalThroughMqtt3
156/175 Test #162: Mqtt5to3AdapterDirectConnectionMinimalThroughMqtt3 ................***Skipped   0.01 sec
[...]
```

</details>

<details>

<summary>114/175: Sigv4SigningTestSimple</summary>

```
[...]
114/175 Test #105: Sigv4SigningTestSimple ............................................Subprocess aborted***Exception:   0.27 sec
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [aws-c-common] - static: libnuma.so failed to load
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - searching process and loaded modules
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - did not find aws-lc symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - did not find boringssl symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - did not find libcrypto 1.0.2 symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - found static libcrypto 1.1.1 HMAC symbols
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - found static libcrypto 1.1.1 EVP_MD symbols
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [libcrypto_resolve] - Compiled with libcrypto OpenSSL 3.3.1 4 Jun 2024, linked to libcrypto OpenSSL 3.3.1 4 Jun 2024
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [tls-handler] - static: Initializing TLS using s2n.
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [tls-handler] - ctx: Based on OS, we detected the default PKI path as /etc/ssl/certs, and ca file as /etc/ssl/certs/ca-certificates.crt
[ERROR] [2024-07-31T10:09:00Z] [000070746f78fcb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/sys_vendor' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[ERROR] [2024-07-31T10:09:00Z] [000070746f78fcb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/product_name' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [aws-c-common] - id=0x70746ec5cf40: virtualization vendor detected as ""
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [aws-c-common] - id=0x70746ec5cf40: virtualization product name detected as " "
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "c5n.18xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "c5n.9xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "c5n.metal".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "p4d.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "p4de.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "p5.48xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "trn1n.32xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [S3General] - id=0x70746ebf80b0: adding platform entry for "trn1.32xlarge".
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9c70: Initializing edge-triggered epoll
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9c70: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9c70: pipe descriptors read 6, write 7.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9c70: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9cc0: Initializing edge-triggered epoll
[INFO] [2024-07-31T10:09:00Z] [000070746ebefb38] [event-loop] - id=0x70746ebf9c70: main loop started
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9cc0: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000070746ebefb38] [event-loop] - id=0x70746ebf9c70: subscribing to events on fd 6
[INFO] [2024-07-31T10:09:00Z] [000070746ebefb38] [event-loop] - id=0x70746ebf9c70: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000070746ebefb38] [event-loop] - id=0x70746ebf9c70: waiting for a maximum of 100000 ms
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9cc0: pipe descriptors read 9, write 10.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9cc0: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d10: Initializing edge-triggered epoll
[INFO] [2024-07-31T10:09:00Z] [000070746eaecb38] [event-loop] - id=0x70746ebf9cc0: main loop started
[TRACE] [2024-07-31T10:09:00Z] [000070746eaecb38] [event-loop] - id=0x70746ebf9cc0: subscribing to events on fd 9
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d10: Eventfd not available, falling back to pipe for cross-thread notification.
[INFO] [2024-07-31T10:09:00Z] [000070746eaecb38] [event-loop] - id=0x70746ebf9cc0: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000070746eaecb38] [event-loop] - id=0x70746ebf9cc0: waiting for a maximum of 100000 ms
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d10: pipe descriptors read 12, write 13.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d10: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d60: Initializing edge-triggered epoll
[INFO] [2024-07-31T10:09:00Z] [000070746e9e9b38] [event-loop] - id=0x70746ebf9d10: main loop started
[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d60: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000070746e9e9b38] [event-loop] - id=0x70746ebf9d10: subscribing to events on fd 12
[INFO] [2024-07-31T10:09:00Z] [000070746e9e9b38] [event-loop] - id=0x70746ebf9d10: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d60: pipe descriptors read 15, write 16.
[TRACE] [2024-07-31T10:09:00Z] [000070746e9e9b38] [event-loop] - id=0x70746ebf9d10: waiting for a maximum of 100000 ms
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [event-loop] - id=0x70746ebf9d60: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [dns] - id=0x70746ec58ca0: Initializing default host resolver with 8 max host entries.
[INFO] [2024-07-31T10:09:00Z] [000070746e8e6b38] [event-loop] - id=0x70746ebf9d60: main loop started
[TRACE] [2024-07-31T10:09:00Z] [000070746e8e6b38] [event-loop] - id=0x70746ebf9d60: subscribing to events on fd 15
[INFO] [2024-07-31T10:09:00Z] [000070746e8e6b38] [event-loop] - id=0x70746ebf9d60: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000070746e8e6b38] [event-loop] - id=0x70746ebf9d60: waiting for a maximum of 100000 ms
[INFO] [2024-07-31T10:09:00Z] [000070746f78fcb0] [channel-bootstrap] - id=0x70746ec2da60: Initializing client bootstrap with event-loop group 0x70746ec2d620
malloc failed to allocate memory[DEBUG] [2024-07-31T10:09:00Z] [000070746f78fcb0] [channel-bootstrap] - id=0x70746ec2da60: acquiring bootstrap reference

115/175 Test #120: Mqtt311WSConnectionWithBasicAuth ..................................***Skipped   0.01 sec
        Start 122: Mqtt311WSConnectionWithHttpProxy
[...]
```

</details>

<details>

<summary>97/175: TestProviderImdsGet</summary>

```
[...]
 97/175 Test  #98: TestProviderImdsGet ...............................................Subprocess aborted***Exception:   0.22 sec
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [aws-c-common] - static: libnuma.so failed to load
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - searching process and loaded modules
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - did not find aws-lc symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - did not find boringssl symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - did not find libcrypto 1.0.2 symbols linked
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - found static libcrypto 1.1.1 HMAC symbols
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - found static libcrypto 1.1.1 EVP_MD symbols
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [libcrypto_resolve] - Compiled with libcrypto OpenSSL 3.3.1 4 Jun 2024, linked to libcrypto OpenSSL 3.3.1 4 Jun 2024
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [tls-handler] - static: Initializing TLS using s2n.
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [tls-handler] - ctx: Based on OS, we detected the default PKI path as /etc/ssl/certs, and ca file as /etc/ssl/certs/ca-certificates.crt
[ERROR] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/sys_vendor' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[ERROR] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [common-io] - static: Failed to open file. path:'/sys/devices/virtual/dmi/id/product_name' mode:'rb' errno:2 aws-error:44(AWS_ERROR_FILE_INVALID_PATH)
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [aws-c-common] - id=0x78ae30353f40: virtualization vendor detected as ""
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [aws-c-common] - id=0x78ae30353f40: virtualization product name detected as " "
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "c5n.18xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "c5n.9xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "c5n.metal".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "p4d.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "p4de.24xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "p5.48xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "trn1n.32xlarge".
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [S3General] - id=0x78ae3033e0b0: adding platform entry for "trn1.32xlarge".
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fc70: Initializing edge-triggered epoll
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fc70: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fc70: pipe descriptors read 6, write 7.
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fc70: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fcc0: Initializing edge-triggered epoll
[INFO] [2024-07-31T10:09:00Z] [000078ae30232b38] [event-loop] - id=0x78ae3033fc70: main loop started
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fcc0: Eventfd not available, falling back to pipe for cross-thread notification.
[TRACE] [2024-07-31T10:09:00Z] [000078ae30232b38] [event-loop] - id=0x78ae3033fc70: subscribing to events on fd 6
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fcc0: pipe descriptors read 9, write 10.
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fcc0: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000078ae30232b38] [event-loop] - id=0x78ae3033fc70: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000078ae30232b38] [event-loop] - id=0x78ae3033fc70: waiting for a maximum of 100000 ms
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd10: Initializing edge-triggered epoll
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd10: Eventfd not available, falling back to pipe for cross-thread notification.
[INFO] [2024-07-31T10:09:00Z] [000078ae3012fb38] [event-loop] - id=0x78ae3033fcc0: main loop started
[TRACE] [2024-07-31T10:09:00Z] [000078ae3012fb38] [event-loop] - id=0x78ae3033fcc0: subscribing to events on fd 9
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd10: pipe descriptors read 12, write 13.
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd10: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000078ae3012fb38] [event-loop] - id=0x78ae3033fcc0: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000078ae3012fb38] [event-loop] - id=0x78ae3033fcc0: waiting for a maximum of 100000 ms
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd60: Initializing edge-triggered epoll
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd60: Eventfd not available, falling back to pipe for cross-thread notification.
[INFO] [2024-07-31T10:09:00Z] [000078ae3002cb38] [event-loop] - id=0x78ae3033fd10: main loop started
[TRACE] [2024-07-31T10:09:00Z] [000078ae3002cb38] [event-loop] - id=0x78ae3033fd10: subscribing to events on fd 12
[TRACE] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd60: pipe descriptors read 15, write 16.
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [event-loop] - id=0x78ae3033fd60: Starting event-loop thread.
[INFO] [2024-07-31T10:09:00Z] [000078ae3002cb38] [event-loop] - id=0x78ae3033fd10: default timeout 100000, and max events to process per tick 100
[TRACE] [2024-07-31T10:09:00Z] [000078ae3002cb38] [event-loop] - id=0x78ae3033fd10: waiting for a maximum of 100000 ms
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [dns] - id=0x78ae303596d0: Initializing default host resolver with 8 max host entries.
[INFO] [2024-07-31T10:09:00Z] [000078ae2ff29b38] [event-loop] - id=0x78ae3033fd60: main loop started
[INFO] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [channel-bootstrap] - id=0x78ae30373a60: Initializing client bootstrap with event-loop group 0x78ae30373680
[TRACE] [2024-07-31T10:09:00Z] [000078ae2ff29b38] [event-loop] - id=0x78ae3033fd60: subscribing to events on fd 15
[INFO] [2024-07-31T10:09:00Z] [000078ae2ff29b38] [event-loop] - id=0x78ae3033fd60: default timeout 100000, and max events to process per tick 100
[DEBUG] [2024-07-31T10:09:00Z] [000078ae30ed5cb0] [channel-bootstrap] - id=0x78ae30373a60: acquiring bootstrap reference
malloc failed to allocate memory[TRACE] [2024-07-31T10:09:00Z] [000078ae2ff29b38] [event-loop] - id=0x78ae3033fd60: waiting for a maximum of 100000 ms

        Start 105: Sigv4SigningTestSimple
 98/175 Test #104: Sigv4SigningTestCreateDestroy .....................................   Passed    0.08 sec
[...]
```

</details>

The test failure of aws-crt-python is here:

</details>

<details>

<summary>FAIL: test_send_frame_exceptions (test_websocket.TestClient.test_send_frame_exceptions)</summary>

```
........................s.............s.ss...ssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss.......ssssssssssssssssssssssssss......Exception in WebSocket on_incoming_frame_begin callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 487, in _on_incoming_frame_begin
    self._on_incoming_frame_begin_cb(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 502, in bad_incoming_frame_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
Exception in WebSocket on_incoming_frame_payload callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 501, in _on_incoming_frame_payload
    self._on_incoming_frame_payload_cb(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 502, in bad_incoming_frame_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
Exception in WebSocket on_incoming_frame_complete callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 519, in _on_incoming_frame_complete
    self._on_incoming_frame_complete_cb(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 502, in bad_incoming_frame_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
.Exception in WebSocket on_connection_setup callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 459, in _on_connection_setup
    self._on_connection_setup_cb(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 334, in bad_setup_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
.Exception in WebSocket on_connection_shutdown callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 474, in _on_connection_shutdown
    self._on_connection_shutdown_cb(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 357, in bad_shutdown_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
..Exception in WebSocket.send_frame on_complete callback
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/awscrt/websocket.py", line 390, in _on_complete
    on_complete(cbdata)
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 480, in bad_completion_callback
    raise RuntimeError("Purposefully raising exception")
RuntimeError: Purposefully raising exception
F.
======================================================================
FAIL: test_send_frame_exceptions (test_websocket.TestClient.test_send_frame_exceptions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/fossdd/aports/community/py3-awscrt/src/aws-crt-python-0.21.2/test/test_websocket.py", line 488, in test_send_frame_exceptions
    with self.assertRaises(Exception) as raises:
AssertionError: Exception not raised
----------------------------------------------------------------------
Ran 305 tests in 21.118s
FAILED (failures=1, skipped=164)
```

</details>

### Reproduction Steps

I couldn't find simple reproducible snippet, but testing the projects above should result in the same behaviour.

### aws-c-http version used

0.8.3-0.8.5

### Compiler and version used

gcc (Alpine 13.2.1_git20240309)

### Operating System and version

Alpine Linux edge

**Comment by waahm7** [comment]
Are you using the latest version of aws-crt-cpp and aws-crt-python? What version of Alpine are you using?

**Comment by fossdd** [comment]
Both libraries were using the latest version, but I also tested
aws-crt-cpp with a older version that once worked with aws-c-http 0.8.2,
and it failed

The full logs are still here (https://gitlab.alpinelinux.org/fossdd/aports/-/jobs/1470570)
where you can also find more information on how it's build etc
(https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/69962)

**Comment by waahm7** [comment]
Can you share more details about your environment, such as the Docker container you are using, the architecture, and the Alpine version, etc.? We have CI running on Alpine, and tests are working fine there. Ideally, some repro steps would be great for figuring out the issue.

**Comment by fossdd** [comment]
Ah, apparently the issue is fixed by upgrading s2n-tls 1.5.0 which got a few days ago. Thanks for your help!
